ledger/src/executor.rs

```diff
@@ -0,0 +1,421 @@
+impl ReplayerHandle {
+    pub fn new(
+        request_sender: Sender<(Sender<ReplayResponse>, ReplayRequest)>,
+        response_sender: Sender<ReplayResponse>,
+        response_receiver: Receiver<ReplayResponse>,
+    ) -> ReplayerHandle {
+        ReplayerHandle {
+            request_sender,
+            response_sender,
+            response_receiver,
+        }
+    }
+
+    pub fn send(
+        &self,
+        request: ReplayRequest,
+    ) -> result::Result<(), SendError<(Sender<ReplayResponse>, ReplayRequest)>> {
+        self.request_sender
+            .send((self.response_sender.clone(), request))
+    }
+
+    pub fn recv_and_drain(&self) -> result::Result<Vec<ReplayResponse>, RecvError> {
+        let mut results = vec![self.response_receiver.recv()?];
+        results.extend(self.response_receiver.try_iter());
+        Ok(results)
+    }
+}
+
+impl Replayer {
+    pub fn new(num_threads: usize) -> Replayer {
+        let (request_sender, request_receiver) = unbounded();
+        let threads = Self::start_replay_threads(num_threads, request_receiver);
+        Replayer {
+            threads,
+            request_sender,
+        }
+    }
+
+    pub fn handle(&self) -> ReplayerHandle {
+        let (response_sender, response_receiver) = unbounded();
+        ReplayerHandle::new(
+            self.request_sender.clone(),
+            response_sender,
+            response_receiver,
+        )
+    }
+
+    pub fn start_replay_threads(
+        num_threads: usize,
+        request_receiver: Receiver<(Sender<ReplayResponse>, ReplayRequest)>,
+    ) -> Vec<JoinHandle<()>> {
+        (0..num_threads)
+            .map(|i| {
+                let request_receiver = request_receiver.clone();
+                Builder::new()
+                    .name(format!("solReplayer-{}", i))
+                    .spawn(move || loop {
+                        match request_receiver.recv() {
+                            Ok((
+                                response_sender,
+                                ReplayRequest {
+                                    tx_and_idx,
+                                    bank,
+                                    transaction_status_sender,
+                                    replay_vote_sender,
+                                    cost_capacity_meter,
+                                    tx_cost,
+                                    log_messages_bytes_limit,
+                                    entry_callback,
+                                    batch_idx,
+                                },
+                            )) => {
+                                let mut timings = ExecuteTimings::default();
+
+                                let txs = vec![tx_and_idx.0];
+                                let mut batch =
+                                    TransactionBatch::new(vec![Ok(())], &bank, Cow::Borrowed(&txs));
+                                batch.set_needs_unlock(false);
```

Comment:
Couple of questions on this as I'm reading through again:
1. We have a bank, why not grab the account locks for this account? Definitely performance wise not grabbing is more efficient, but it'd probably be better to do this replay separation, then removing locking in separate changes. Once we're convinced we don't need the locks anymore.
2. I'm not entirely convinced this can actually get rid of the locks. If we are replaying active banks concurrently, is there something I'm not aware of that prevents those from touching the same account(s)?

Reply:
yes, we can do that. if its locked, perhaps we should continually spin until its unlocked, perhaps with timeout or assert?

Reply:
alternatively, could grab the lock in caller?

Reply:
I feel it's more intuitive to have the executor do the locking. Yeah we could either just spin until it's unlocked, or even add it to some queue of blocked work that is rechecked if unblocked every loop.

I imagine spinning is probably fine though, since it sounds like you've not hit this yet and is probably a rare edge case.